src/AspNetCore/Src/applicationmanager.cxx

```diff
@@ -480,19 +480,18 @@ APPLICATION_MANAGER::DoRecycleApplication(
 // static
 VOID
 APPLICATION_MANAGER::RecycleApplication(
-    _In_ APPLICATION_INFO *     pEntry,
-    _In_ APP_HOSTING_MODEL      hostingModel
+    _In_ APPLICATION_INFO *     pEntry
 )
 {
-
     APPLICATION* pApplication = pEntry->QueryApplication();
+    DBG_ASSERT(pApplication != NULL);
 
     // Reference the application first
     pApplication->ReferenceApplication();
 
-    if (hostingModel == APP_HOSTING_MODEL::HOSTING_OUT_PROCESS)
+    if (pApplication->QueryConfig()->QueryHostingModel() == HOSTING_OUT_PROCESS)
```

Comment:
Actually, can you add a comment explaining why we need to clear the application only for out of process?

Reply:
done